compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp

```diff
@@ -528,6 +528,149 @@ wrapOpInDispatchRegion(RewriterBase &rewriter, Operation *op) {
   return newRegionOp;
 }
 
+FailureOr<Operation *> hoistOutOfDispatch(RewriterBase &rewriter,
```

Comment:
@IanWood1 I think you have more context about flow dispatches. Could you help review the function (or the PR) and see if we can simplify the code, if you are available? Having another eyes on reviews sounds better to me.